### ISSUE TITLE
Hyprbars can't build on NixOS

### ISSUE THREAD

**Issue by dasehak** [hyprwm/hyprland-plugins#189]
log:
```
hyprbars> Running phase: unpackPhase
hyprbars> unpacking source archive /nix/store/nnvcdzryal3v5p8pc0fyq7xfn18akdvp-hyprbars
hyprbars> source root is hyprbars
hyprbars> Running phase: patchPhase
hyprbars> Running phase: updateAutotoolsGnuConfigScriptsPhase
hyprbars> Running phase: configurePhase
hyprbars> fixing cmake files...
hyprbars> cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/jf2jpch7k7dkf0j2kk9kpfdadidd9dm4-hyprbars-0.1/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/jf2jpch7k7dkf0j2kk9kpfdadidd9dm4-hyprbars-0.1/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/jf2jpch7k7dkf0j2kk9kpfdadidd9dm4-hyprbars-0.1/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/jf2jpch7k7dkf0j2kk9kpfdadidd9dm4-hyprbars-0.1/share/doc/hyprbars -DCMAKE_INSTALL_INFODIR=/nix/store/jf2jpch7k7dkf0j2kk9kpfdadidd9dm4-hyprbars-0.1/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/jf2jpch7k7dkf0j2kk9kpfdadidd9dm4-hyprbars-0.1/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/jf2jpch7k7dkf0j2kk9kpfdadidd9dm4-hyprbars-0.1/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/jf2jpch7k7dkf0j2kk9kpfdadidd9dm4-hyprbars-0.1/include -DCMAKE_INSTALL_SBINDIR=/nix/store/jf2jpch7k7dkf0j2kk9kpfdadidd9dm4-hyprbars-0.1/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/jf2jpch7k7dkf0j2kk9kpfdadidd9dm4-hyprbars-0.1/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/jf2jpch7k7dkf0j2kk9kpfdadidd9dm4-hyprbars-0.1/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/mpm3i0sbqc9svfch6a17179fs64dz2kv-gcc-wrapper-13.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/mpm3i0sbqc9svfch6a17179fs64dz2kv-gcc-wrapper-13.3.0/bin/ranlib -DCMAKE_AR=/nix/store/mpm3i0sbqc9svfch6a17179fs64dz2kv-gcc-wrapper-13.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/jf2jpch7k7dkf0j2kk9kpfdadidd9dm4-hyprbars-0.1  
hyprbars> -- The C compiler identification is GNU 13.3.0
hyprbars> -- The CXX compiler identification is GNU 13.3.0
hyprbars> -- Detecting C compiler ABI info
hyprbars> -- Detecting C compiler ABI info - done
man-cache> Updating index cache for path `/nix/store/zwpn6y8ny4hyyarkidnv8xvw9gs63jjz-man-paths/share/man/man1'. Wait...
man-cache> gzip: stdin: not in gzip format
man-cache> 
man-cache> gzip: stdin: not in gzip format
man-cache> /nix/store/9xsvcq63h0jys7pc0nx3q8mg25ppbi0g-man-db-2.12.1/bin/mandb: warning: /nix/store/zyr8rwj923xxf1267phhkchdyk810bax-waypaper-2.1.2/share/man/man1/waypaper.1.gz: whatis parse for waypaper(1) failed
hyprbars> -- Check for working C compiler: /nix/store/mpm3i0sbqc9svfch6a17179fs64dz2kv-gcc-wrapper-13.3.0/bin/gcc - skipped
hyprbars> -- Detecting C compile features
hyprbars> -- Detecting C compile features - done
hyprbars> -- Detecting CXX compiler ABI info
hyprbars> -- Detecting CXX compiler ABI info - done
hyprbars> -- Check for working CXX compiler: /nix/store/mpm3i0sbqc9svfch6a17179fs64dz2kv-gcc-wrapper-13.3.0/bin/g++ - skipped
hyprbars> -- Detecting CXX compile features
hyprbars> -- Detecting CXX compile features - done
hyprbars> -- Found PkgConfig: /nix/store/2hfzaqv42iwrpk3ya24cnjsklz6f68lw-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
hyprbars> -- Checking for modules 'hyprland;libdrm;libinput;libudev;pangocairo;pixman-1;wayland-server;xkbcommon'
hyprbars> --   Found hyprland, version 0.41.1
hyprbars> --   Found libdrm, version 2.4.120
hyprbars> --   Found libinput, version 1.25.0
hyprbars> --   Found libudev, version 255
hyprbars> --   Found pangocairo, version 1.52.2
hyprbars> --   Found pixman-1, version 0.43.4
hyprbars> --   Found wayland-server, version 1.22.0
hyprbars> --   Found xkbcommon, version 1.7.0
hyprbars> -- Configuring done (2.1s)
hyprbars> -- Generating done (0.0s)
hyprbars> CMake Warning:
hyprbars>   Manually-specified variables were not used by the project:
hyprbars> 
hyprbars>     BUILD_TESTING
hyprbars>     CMAKE_EXPORT_NO_PACKAGE_REGISTRY
hyprbars>     CMAKE_INSTALL_BINDIR
hyprbars>     CMAKE_INSTALL_DOCDIR
hyprbars>     CMAKE_INSTALL_INCLUDEDIR
hyprbars>     CMAKE_INSTALL_INFODIR
hyprbars>     CMAKE_INSTALL_LIBEXECDIR
hyprbars>     CMAKE_INSTALL_LOCALEDIR
hyprbars>     CMAKE_INSTALL_MANDIR
hyprbars>     CMAKE_INSTALL_OLDINCLUDEDIR
hyprbars>     CMAKE_INSTALL_SBINDIR
hyprbars>     CMAKE_POLICY_DEFAULT_CMP0025
hyprbars> 
hyprbars> 
hyprbars> -- Build files have been written to: /build/hyprbars/build
hyprbars> cmake: enabled parallel building
hyprbars> cmake: enabled parallel installing
hyprbars> Running phase: buildPhase
hyprbars> build flags: -j24
hyprbars> [1/4] Building CXX object CMakeFiles/hyprbars.dir/CMakeFiles/3.29.3/CompilerIdCXX/CMakeCXXCompilerId.cpp.o
man-cache> Updating index cache for path `/nix/store/zwpn6y8ny4hyyarkidnv8xvw9gs63jjz-man-paths/share/man/man8'. Wait...done.
man-cache> BDB3028 /nix/store/s59nbpa388z1i605hdv0xpgv3m396q8s-man-cache/5: unable to flush: No such file or directory
man-cache> Processing manual pages under /nix/store/zwpn6y8ny4hyyarkidnv8xvw9gs63jjz-man-paths/share/man/it...
man-cache> Updating index cache for path `/nix/store/zwpn6y8ny4hyyarkidnv8xvw9gs63jjz-man-paths/share/man/it/man8'. Wait...done.
man-cache> BDB3028 /nix/store/s59nbpa388z1i605hdv0xpgv3m396q8s-man-cache/it/5: unable to flush: No such file or directory
man-cache> 8 man subdirectories contained newer manual pages.
man-cache> 377 manual pages were added.
hm_.manpath> building '/nix/store/n454l0c4bn851an5d9hxdc7sz388i8im-hm_.manpath.drv'
hyprbars> [2/4] Building CXX object CMakeFiles/hyprbars.dir/barDeco.cpp.o
hyprbars> FAILED: CMakeFiles/hyprbars.dir/barDeco.cpp.o 
hyprbars> /nix/store/mpm3i0sbqc9svfch6a17179fs64dz2kv-gcc-wrapper-13.3.0/bin/g++ -Dhyprbars_EXPORTS -isystem /nix/store/sbz5qy9whqwz2j78k0fr7v53mm3axgmp-hyprland-0.41.1+date=2024-06-23_e09addf-dev/include/hyprland/protocols -isystem /nix/store/sbz5qy9whqwz2j78k0fr7v53mm3axgmp-hyprland-0.41.1+date=2024-06-23_e09addf-dev/include/hyprland -isystem /nix/store/sbz5qy9whqwz2j78k0fr7v53mm3axgmp-hyprland-0.41.1+date=2024-06-23_e09addf-dev/include/hyprland/wlr -isystem /nix/store/177z07c2gkqvdni3wrv68kxbgxg5jc5j-cairo-1.18.0-dev/include/cairo -isystem /nix/store/2nrdpdpqsi94arfv3xh9pa3ywqpa38ry-glib-2.80.2-dev/include/glib-2.0 -isystem /nix/store/nm9608b5y801fq2p73nl7k80z8kcbmh2-glib-2.80.2/lib/glib-2.0/include -isystem /nix/store/5f2704za85mzq80m18pb6bc7fy77bzsg-libdrm-2.4.120-dev/include/libdrm -isystem /nix/store/63b2n1iada8lvmafhrg139dampi819zg-pango-1.52.2-dev/include/pango-1.0 -isystem /nix/store/jw2x10k0lnfw5fqi1962bha7qnd6f6kf-harfbuzz-8.4.0-dev/include/harfbuzz -O3 -DNDEBUG -std=gnu++23 -fPIC -MD -MT CMakeFiles/hyprbars.dir/barDeco.cpp.o -MF CMakeFiles/hyprbars.dir/barDeco.cpp.o.d -o CMakeFiles/hyprbars.dir/barDeco.cpp.o -c /build/hyprbars/barDeco.cpp
hyprbars> In file included from /build/hyprbars/barDeco.cpp:1:
hyprbars> /build/hyprbars/barDeco.hpp:40:5: error: 'SWindowDecorationExtents' does not name a type
hyprbars>    40 |     SWindowDecorationExtents m_seExtents;
hyprbars>       |     ^~~~~~~~~~~~~~~~~~~~~~~~
hyprbars> /build/hyprbars/barDeco.cpp: In destructor 'virtual CHyprBar::~CHyprBar()':
hyprbars> /build/hyprbars/barDeco.cpp:27:36: warning: 'bool HyprlandAPI::unregisterCallback(void*, Hyprutils::Memory::CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >)' is deprecated [-Wdeprecated-declarations]
hyprbars>    27 |     HyprlandAPI::unregisterCallback(PHANDLE, m_pMouseButtonCallback);
hyprbars>       |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hyprbars> In file included from /build/hyprbars/globals.hpp:3,
hyprbars>                  from /build/hyprbars/barDeco.hpp:8:
hyprbars> /nix/store/sbz5qy9whqwz2j78k0fr7v53mm3axgmp-hyprland-0.41.1+date=2024-06-23_e09addf-dev/include/hyprland/src/plugins/PluginAPI.hpp:155:33: note: declared here
hyprbars>   155 |     APICALL [[deprecated]] bool unregisterCallback(HANDLE handle, SP<HOOK_CALLBACK_FN> fn);
hyprbars>       |                                 ^~~~~~~~~~~~~~~~~~
hyprbars> /build/hyprbars/barDeco.cpp:28:36: warning: 'bool HyprlandAPI::unregisterCallback(void*, Hyprutils::Memory::CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >)' is deprecated [-Wdeprecated-declarations]
hyprbars>    28 |     HyprlandAPI::unregisterCallback(PHANDLE, m_pMouseMoveCallback);
hyprbars>       |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hyprbars> /nix/store/sbz5qy9whqwz2j78k0fr7v53mm3axgmp-hyprland-0.41.1+date=2024-06-23_e09addf-dev/include/hyprland/src/plugins/PluginAPI.hpp:155:33: note: declared here
hyprbars>   155 |     APICALL [[deprecated]] bool unregisterCallback(HANDLE handle, SP<HOOK_CALLBACK_FN> fn);
hyprbars>       |                                 ^~~~~~~~~~~~~~~~~~
hyprbars> /build/hyprbars/barDeco.cpp: In member function 'virtual SDecorationPositioningInfo CHyprBar::getPositioningInfo()':
hyprbars> /build/hyprbars/barDeco.cpp:41:42: warning: narrowing conversion of '(((CHyprBar*)this)->CHyprBar::m_bHidden ? 0 : (*(long int*)(*(long int* const*)PHEIGHT)))' from 'long int' to 'int' [-Wnarrowing]
hyprbars>    41 |     info.desiredExtents = {{0, m_bHidden ? 0 : **PHEIGHT}, {0, 0}};
hyprbars>       |                                ~~~~~~~~~~^~~~~~~~~~~~~~~
hyprbars> /build/hyprbars/barDeco.cpp: In member function 'void CHyprBar::onMouseDown(SCallbackInfo&, IPointer::SButtonEvent)':
hyprbars> /build/hyprbars/barDeco.cpp:83:17: warning: comparison between 'enum wl_pointer_button_state' and 'enum wlr_button_state' [-Wenum-compare]
hyprbars>    83 |     if (e.state != WLR_BUTTON_PRESSED) {
hyprbars>       |         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
hyprbars> /build/hyprbars/barDeco.cpp:113:70: warning: narrowing conversion of '*(long int*)(*(long int* const*)PHEIGHT)' from 'long int' to 'int' [-Wnarrowing]
hyprbars>   113 |         const auto BARBUF     = Vector2D{(int)assignedBoxGlobal().w, **PHEIGHT};
hyprbars>       |                                                                      ^~~~~~~~~
hyprbars> /build/hyprbars/barDeco.cpp: In member function 'virtual void CHyprBar::draw(CMonitor*, float)':
hyprbars> /build/hyprbars/barDeco.cpp:418:5: error: 'm_seExtents' was not declared in this scope
hyprbars>   418 |     m_seExtents = {{0, **PHEIGHT}, {}};
hyprbars>       |     ^~~~~~~~~~~
hyprbars> /build/hyprbars/barDeco.cpp:478:51: warning: narrowing conversion of '(int)(double)BARBUF.Hyprutils::Math::Vector2D::x' from 'int' to 'double' [-Wnarrowing]
hyprbars>   478 |     CBox textBox = {titleBarBox.x, titleBarBox.y, (int)BARBUF.x, (int)BARBUF.y};
hyprbars>       |                                                   ^~~~~~~~~~~~~
hyprbars> /build/hyprbars/barDeco.cpp:478:66: warning: narrowing conversion of '(int)(double)BARBUF.Hyprutils::Math::Vector2D::y' from 'int' to 'double' [-Wnarrowing]
hyprbars>   478 |     CBox textBox = {titleBarBox.x, titleBarBox.y, (int)BARBUF.x, (int)BARBUF.y};
hyprbars>       |                                                                  ^~~~~~~~~~~~~
hyprbars> [3/4] Building CXX object CMakeFiles/hyprbars.dir/main.cpp.o
hyprbars> FAILED: CMakeFiles/hyprbars.dir/main.cpp.o 
hyprbars> /nix/store/mpm3i0sbqc9svfch6a17179fs64dz2kv-gcc-wrapper-13.3.0/bin/g++ -Dhyprbars_EXPORTS -isystem /nix/store/sbz5qy9whqwz2j78k0fr7v53mm3axgmp-hyprland-0.41.1+date=2024-06-23_e09addf-dev/include/hyprland/protocols -isystem /nix/store/sbz5qy9whqwz2j78k0fr7v53mm3axgmp-hyprland-0.41.1+date=2024-06-23_e09addf-dev/include/hyprland -isystem /nix/store/sbz5qy9whqwz2j78k0fr7v53mm3axgmp-hyprland-0.41.1+date=2024-06-23_e09addf-dev/include/hyprland/wlr -isystem /nix/store/177z07c2gkqvdni3wrv68kxbgxg5jc5j-cairo-1.18.0-dev/include/cairo -isystem /nix/store/2nrdpdpqsi94arfv3xh9pa3ywqpa38ry-glib-2.80.2-dev/include/glib-2.0 -isystem /nix/store/nm9608b5y801fq2p73nl7k80z8kcbmh2-glib-2.80.2/lib/glib-2.0/include -isystem /nix/store/5f2704za85mzq80m18pb6bc7fy77bzsg-libdrm-2.4.120-dev/include/libdrm -isystem /nix/store/63b2n1iada8lvmafhrg139dampi819zg-pango-1.52.2-dev/include/pango-1.0 -isystem /nix/store/jw2x10k0lnfw5fqi1962bha7qnd6f6kf-harfbuzz-8.4.0-dev/include/harfbuzz -O3 -DNDEBUG -std=gnu++23 -fPIC -MD -MT CMakeFiles/hyprbars.dir/main.cpp.o -MF CMakeFiles/hyprbars.dir/main.cpp.o.d -o CMakeFiles/hyprbars.dir/main.cpp.o -c /build/hyprbars/main.cpp
hyprbars> In file included from /build/hyprbars/main.cpp:10:
hyprbars> /build/hyprbars/barDeco.hpp:40:5: error: 'SWindowDecorationExtents' does not name a type
hyprbars>    40 |     SWindowDecorationExtents m_seExtents;
hyprbars>       |     ^~~~~~~~~~~~~~~~~~~~~~~~
hyprbars> ninja: build stopped: subcommand failed.

```

**Comment by fufexan** [comment]
Not Nix-specific, it doesn't build anywhere. Needs to be updated.

**Comment by fufexan** [comment]
Fixed in db46f6a.
